tests/lib/lte_lc_api/src/lte_lc_api_test.c

```diff
@@ -283,6 +293,214 @@ void test_lte_lc_connect_async_null(void)
+	strcpy(at_notif, "+CEREG: 0\r\n");
+	at_monitor_dispatch(at_notif);
+}
+
+void test_lte_lc_connect_fallback(void)
+{
+	int ret;
+
+	/* AT commands triggered by lte_lc_connect() */
+	__mock_nrf_modem_at_scanf_ExpectAndReturn(
+		"AT+CEREG?", "+CEREG: %*u,%hu,%*[^,],\"%x\",", 1);
+	__mock_nrf_modem_at_scanf_ReturnVarg_uint16(LTE_LC_NW_REG_NOT_REGISTERED);
+
+	__mock_nrf_modem_at_scanf_ExpectAndReturn(
+		"AT+CFUN?", "+CFUN: %hu", 1);
+	__mock_nrf_modem_at_scanf_ReturnVarg_uint16(LTE_LC_FUNC_MODE_OFFLINE);
+
+	__mock_nrf_modem_at_printf_ExpectAndReturn(
+		"AT%XSYSTEMMODE=1,0,1,0", 0);
+
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CEREG=5", 0);
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CSCON=1", 0);
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=1", 0);
+
+	/* Fallback to NB-IoT */
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=4", 0);
+
+	__mock_nrf_modem_at_printf_ExpectAndReturn(
+		"AT%XSYSTEMMODE=0,1,1,0", 0);
+
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CEREG=5", 0);
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CSCON=1", 0);
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=1", 0);
+
+	static const char xmonitor_resp[] =
+		"%XMONITOR: 5,\"Operator\",\"OP\",\"20065\",\"003F\",9,20,\"0013BEEF\","
+		"334,6200,66,44,\"\","
+		"\"11100000\",\"00101000\",\"11100000\"";
+	__cmock_nrf_modem_at_cmd_ExpectAndReturn(NULL, 0, "AT%%XMONITOR", 0);
+	__cmock_nrf_modem_at_cmd_IgnoreArg_buf();
+	__cmock_nrf_modem_at_cmd_IgnoreArg_len();
+	__cmock_nrf_modem_at_cmd_ReturnArrayThruPtr_buf(
+		(char *)xmonitor_resp, sizeof(xmonitor_resp));
+
+	/* AT commands triggered by lte_lc_offline() */
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=4", 0);
+
+	lte_lc_callback_count_expected = 7;
+
+	/* NB-IoT registration */
+	test_event_data[0].type = LTE_LC_EVT_NW_REG_STATUS;
+	test_event_data[0].nw_reg_status = LTE_LC_NW_REG_REGISTERED_ROAMING;
+
+	test_event_data[1].type = LTE_LC_EVT_CELL_UPDATE;
+	test_event_data[1].cell.id = 0x13beef;
+	test_event_data[1].cell.tac = 0x3f;
+
+	test_event_data[2].type = LTE_LC_EVT_LTE_MODE_UPDATE;
+	test_event_data[2].lte_mode = LTE_LC_LTE_MODE_NBIOT;
+
+	test_event_data[3].type = LTE_LC_EVT_PSM_UPDATE;
+	test_event_data[3].psm_cfg.tau = 28800;
+	test_event_data[3].psm_cfg.active_time = -1;
+
+	/* De-registration */
+	test_event_data[4].type = LTE_LC_EVT_NW_REG_STATUS;
+	test_event_data[4].nw_reg_status = LTE_LC_NW_REG_NOT_REGISTERED;
+
+	test_event_data[5].type = LTE_LC_EVT_CELL_UPDATE;
+	test_event_data[5].cell.id = LTE_LC_CELL_EUTRAN_ID_INVALID;
+	test_event_data[5].cell.tac = LTE_LC_CELL_TAC_INVALID;
+
+	test_event_data[6].type = LTE_LC_EVT_LTE_MODE_UPDATE;
+	test_event_data[6].lte_mode = LTE_LC_LTE_MODE_NONE;
+
+	/* Schedule +CEREG notification to be dispatched after timeout to trigger fallback */
+	strcpy(at_notif, "+CEREG: 5,\"003F\",\"0013BEEF\",9\r\n");
+	k_work_schedule(&at_notif_dispatch_work, K_SECONDS(CONFIG_LTE_NETWORK_TIMEOUT + 1));
```

Comment:
Maybe set CONFIG_LTE_NETWORK_TIMEOUT=1 in prj.conf. Unfortunately it cannot be made smaller.
But +1 could be made smaller and this could be like 1.3 or 1.5 seconds overall.

Reply:
Set the timeout to 1 second and optimized the sleep times.